jsapp/js/components/reports/chartColorsPicker.component.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import bem from 'js/bem';
+import {REPORT_COLOR_SETS} from './reportsConstants';
+import type {ReportStyle, ReportColorSet} from './reportsConstants';
+
+interface ChartColorsPickerProps {
+  onChange: (params: {default: boolean}, value: {report_colors: string[]}) => void;
+  defaultStyle: ReportStyle;
+}
+
+export default function ChartColorsPicker(props: ChartColorsPickerProps) {
+  function defaultReportColorsChange(value: number) {
+    let newColors = REPORT_COLOR_SETS[0].colors;
+    if (REPORT_COLOR_SETS[value]?.colors) {
+      newColors = REPORT_COLOR_SETS[value].colors;
+    }
+
+    props.onChange({default: true}, {report_colors: newColors});
+  }
+
+  // Not sure why this is called "is default", and not simply "is active". This
+  // needs some more investigation.
```

Comment:
Digging deeper into this, turned out that both `ChartColorsPicker` and `ChartTypePicker` were using these weird names with "default" word, but in fact these were more like "current thing being edited right now". So I ended up renaming some properties and removing some more code that was not being used.

Reply:
I did find a bug:

1. Create custom report and for the colors choose the gray palette
2. Open single question report settings and modify colors (overrides the gray)
3. Open single question report settings and use Reset button
4. I expected to have the gray again, but instead I got some other colors (from `DEFAULT_MINIMAL_REPORT_STYLE`?)

Reply:
I fixed the bug. It doesn't behave ideally, but it's consistent with what we have in `beta` right now. What I mean is that:

1. Create custom report and for the colors choose the gray palette
2. Open single question report settings and modify colors
  - You can notice that no colors are selected
  - It makes sense, as the "override for single question" doesn't exist yet
  - Ideally we should see the gray colors being selected in the UI, but the code in the background should be able to know the difference. I think it would require even more refactor, so I'm not going further with this

Reply:
I also went and added more comments and renamed bunch of components (and I am sorry for that, as I understand it might bring on some confusion), to names that make more sense and would be beneficial for future:
- `CustomReportForm` → `CustomReportEditor`
- `ChartColorsPicker` → `ReportColorsEditor`
- `ChartTypePicker` → `ReportTypeEditor`
- `QuestionGraphSettings` → `ReportStyleSettingsSingleQuestion`